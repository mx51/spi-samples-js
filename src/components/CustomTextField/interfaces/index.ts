import { IFormEventValue } from '../../PairPage/PairForm/interfaces';

type Partial<T> = {
  [P in keyof T]?: T[P];
};

type InputProps = {
  endAdornment?: React.ReactNode;
};

type TInputProps = {
  maxLength?: number;
};

export interface ICustomTextField {
  disabled?: boolean;
  error?: boolean;
  fullWidth?: boolean;
  helperText?: string;
  id?: string;
  inputProps?: Partial<TInputProps>;
  InputProps?: Partial<InputProps>;
  label?: string;
  margin?: 'dense' | 'none';
  onBlur?: (event: IFormEventValue) => void;
  onChange?: (event: IFormEventValue) => void;
  required?: boolean;
  value?: string;
  variant?: 'filled' | 'standard' | 'outlined' | undefined;
}

export interface IErrorInputAdornment {
  isValid: boolean;
}
