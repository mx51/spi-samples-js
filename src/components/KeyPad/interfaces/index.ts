export interface IKeyPadProps {
  title: string;
  subtitle: string;
  defaultAmount: number;
  onAmountChange: (amount: number) => void;
  open: boolean;
  onClose: () => void;
}
