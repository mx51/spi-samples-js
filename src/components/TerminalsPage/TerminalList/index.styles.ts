import { createStyles, makeStyles, Theme } from '@material-ui/core';
import {
  CHIP_BACKGROUND_SUCCESS_COLOR,
  CHIP_FONT_SUCCESS_COLOR,
} from '../../../definitions/constants/themeStylesConfigs';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    table: {
      marginTop: theme.spacing(2),
    },
    chipConnected: {
      background: `${CHIP_BACKGROUND_SUCCESS_COLOR}`,
      color: `${CHIP_FONT_SUCCESS_COLOR}`,
    },
  })
);

export default useStyles;
