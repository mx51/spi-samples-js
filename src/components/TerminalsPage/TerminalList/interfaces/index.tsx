export interface ITerminalRecord {
  posId: string;
  pairingStatus: PairingStatus;
  deviceAddress: string;
  serialNumber: string;
  status: string;
}
export interface ITerminalsRecords {
  terminals: Array<ITerminalRecord>;
}

export enum PairingStatus {
  Connecting,
  Connected,
  Disconnected,
  Idle,
}
