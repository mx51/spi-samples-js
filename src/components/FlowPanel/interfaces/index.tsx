import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';

export interface IFlowPanel {
  flow: boolean;
  receiptToggle?: boolean;
}

export interface IFlowPanelState {
  flow: boolean;
  terminal?: ITerminalProps | null;
}
