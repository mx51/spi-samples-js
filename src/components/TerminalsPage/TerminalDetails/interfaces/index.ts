import React from 'react';
import { ITerminalProps } from '../../../../redux/reducers/TerminalSlice/interfaces';

export interface IReceiptToggles {
  settlement: boolean;
  settlementEnquiry: boolean;
}

export interface TabPanelProps {
  children: React.ReactNode;
  index: number;
  subtitle: string;
  title: string;
  value: number;
  receiptToggle?: IReceiptToggles;
  terminal?: ITerminalProps | null;
}

export interface ITerminalConfigurationConstants {
  title: string;
  content: string;
}

export interface IAboutTerminal {
  receiptToggle: IReceiptToggles;
  setReceiptToggle: (receiptToggle: IReceiptToggles) => void;
  terminal: ITerminalProps;
}
