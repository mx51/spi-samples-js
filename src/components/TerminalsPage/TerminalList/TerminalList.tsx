import { Chip, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@material-ui/core';
import React from 'react';
import { useHistory } from 'react-router-dom';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';
import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';
import { getTitleFromStatus } from '../../../utils/common/pair/pairStatusHelpers';
import TerminalActionMenu from '../../TerminalActionMenu';
import useStyles from './index.styles';
import { ITerminalList } from './interfaces';
import { PATH_TERMINALS } from '../../../definitions/constants/routerConfigs';

function TerminalList({ terminals }: ITerminalList): React.ReactElement {
  const classes = useStyles();
  const history = useHistory();

  const chipStyles = (status: string) => {
    switch (status) {
      case SPI_PAIR_STATUS.PairedConnected:
        return classes.chipConnected;
      case SPI_PAIR_STATUS.PairedConnecting:
        return classes.chipPending;
      case SPI_PAIR_STATUS.Unpaired:
        return classes.chipUnpaired;
      default:
        return classes.chipUnpaired;
    }
  };

  const goToTerminalDetails = (path: string) => {
    history.push(path);
  };

  return (
    <TableContainer component={Paper} className={classes.table} elevation={0}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>POS ID</TableCell>
            <TableCell>Pairing status</TableCell>
            <TableCell>EFTPOS address</TableCell>
            <TableCell>Serial number</TableCell>
            <TableCell />
          </TableRow>
        </TableHead>
        <TableBody>
          {terminals.map((terminal: ITerminalProps) => (
            <TableRow id={`terminal_${terminal.serialNumber}`} key={`terminal_${terminal.deviceAddress}`}>
              <TableCell
                className={classes.link}
                onClick={() => goToTerminalDetails(`${PATH_TERMINALS}/${terminal.serialNumber}`)}
                scope="row"
              >
                {terminal.posId}
              </TableCell>
              <TableCell
                className={classes.link}
                onClick={() => goToTerminalDetails(`${PATH_TERMINALS}/${terminal.serialNumber}`)}
              >
                <Chip
                  size="small"
                  label={getTitleFromStatus(terminal.status, terminal.reconnecting)}
                  className={chipStyles(terminal.status)}
                />
              </TableCell>
              <TableCell
                className={classes.link}
                onClick={() => goToTerminalDetails(`${PATH_TERMINALS}/${terminal.serialNumber}`)}
              >
                {terminal.deviceAddress}
              </TableCell>
              <TableCell
                className={classes.link}
                onClick={() => goToTerminalDetails(`${PATH_TERMINALS}/${terminal.serialNumber}`)}
              >
                {terminal.serialNumber}
              </TableCell>
              <TableCell>
                <TerminalActionMenu terminal={terminal} />
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default TerminalList;
