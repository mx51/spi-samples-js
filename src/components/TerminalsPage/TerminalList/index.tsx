import React from 'react';
import {
  Chip,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableFooter,
  TableHead,
  TableRow,
} from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';
import { PATH_TERMINALS } from '../../../definitions/constants/routerConfigs';
import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';
import useStyles from './index.styles';
import { ITerminalList } from './interfaces';

function TerminalList({ terminals }: ITerminalList): React.ReactElement {
  const classes = useStyles();
  const history = useHistory();

  const goToTerminalDetails = (path: string) => {
    history.push(path);
  };

  const chipStyles = (status: string) => {
    switch (status) {
      case SPI_PAIR_STATUS.PairedConnected:
        return classes.chipConnected;
      case SPI_PAIR_STATUS.PairedConnecting:
        return classes.chipPending;
      case SPI_PAIR_STATUS.Unpaired:
        return classes.chipUnpaired;
      default:
        return classes.chipUnpaired;
    }
  };

  return (
    <TableContainer component={Paper} className={classes.table} elevation={0}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>POS ID</TableCell>
            <TableCell>Pairing status</TableCell>
            <TableCell>EFTPOS address</TableCell>
            <TableCell>Serial number</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {terminals.map((terminal: ITerminalProps) => (
            <TableRow
              className={terminal.status !== SPI_PAIR_STATUS.PairedConnected ? classes.unclickable : classes.link}
              id={`terminal_${terminal.serialNumber}`}
              key={`terminal_${terminal.deviceAddress}`}
              onClick={() => goToTerminalDetails(`${PATH_TERMINALS}/${terminal.serialNumber}`)}
            >
              <TableCell scope="row">{terminal.posId}</TableCell>
              <TableCell>
                <Chip size="small" label={terminal.status} className={chipStyles(terminal.status)} />
              </TableCell>
              <TableCell>{terminal.deviceAddress}</TableCell>
              <TableCell>{terminal.serialNumber}</TableCell>
            </TableRow>
          ))}
        </TableBody>
        <TableFooter>
          <TableRow>
            <TableCell colSpan={4} align="center">
              Page 1
            </TableCell>
          </TableRow>
        </TableFooter>
      </Table>
    </TableContainer>
  );
}

export default TerminalList;
