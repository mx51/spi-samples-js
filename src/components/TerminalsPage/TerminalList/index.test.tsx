import React from 'react';
import { cleanup, fireEvent } from '@testing-library/react';
import TerminalList from '.';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';
import { defaultAAR } from '../../../definitions/constants/spiConfigs';
import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';
import { updatePairingStatus } from '../../../redux/reducers/TerminalSlice/terminalsSlice';
import spiService from '../../../services/spiService';
import mockWithRedux, { defaultMockTerminals, mockTerminalInstanceId } from '../../../utils/tests/common';

jest.mock('../../../services/spiService');

const mockTerminalConfig: ITerminalProps = {
  ...defaultMockTerminals[mockTerminalInstanceId],
  deviceAddress: defaultAAR,
  posId: 'test',
  serialNumber: mockTerminalInstanceId,
  status: SPI_PAIR_STATUS.Unpaired,
};

describe('Test <TerminalList />', () => {
  afterEach(cleanup);

  afterAll(() => {
    jest.clearAllMocks();
  });

  test('should match TerminalList snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<TerminalList terminals={[mockTerminalConfig]} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  test('should be able to click current un-pair terminal instance', () => {
    // Arrange
    mockWithRedux(<TerminalList terminals={[mockTerminalConfig]} />);
    const terminalRecord = document.querySelector(`#terminal_${mockTerminalInstanceId}`) as Element;

    // Act
    const dispatch = jest.fn().mockImplementation(() =>
      updatePairingStatus({
        id: mockTerminalInstanceId,
        status: SPI_PAIR_STATUS.Unpaired,
      })
    );

    fireEvent.click(terminalRecord);

    spiService.readTerminalInstance = jest.fn().mockReturnValue(() => ({
      spiClient: {
        Unpair: jest.fn(),
      },
    }));

    spiService.spiTerminalUnPair = jest
      .fn()
      .mockImplementation(() => spiService.readTerminalInstance(mockTerminalInstanceId));

    spiService.spiTerminalUnPair(mockTerminalInstanceId);
    dispatch();

    // Assert
    expect(spiService.spiTerminalUnPair).toBeDefined();
    expect(spiService.spiTerminalUnPair).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalled();
  });
});
