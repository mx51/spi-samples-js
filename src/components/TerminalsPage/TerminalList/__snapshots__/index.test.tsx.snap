// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TerminalList /> should match TerminalList snapshot test 1`] = `
<div>
  <div
    class="MuiPaper-root MuiTableContainer-root makeStyles-table-1 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            POS ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Pairing status
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            EFTPOS address
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Serial number
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root makeStyles-link-3"
          id="terminal_123-123-123"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
            scope="row"
          >
            test
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="MuiChip-root makeStyles-chipConnected-2 MuiChip-sizeSmall"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall"
              >
                Unpaired
              </span>
            </div>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            https://mx51.aar.mock.link
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            123-123-123
          </td>
        </tr>
      </tbody>
      <tfoot
        class="MuiTableFooter-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-footer"
        >
          <td
            class="MuiTableCell-root MuiTableCell-footer MuiTableCell-alignCenter"
            colspan="4"
          >
            Page 1
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
